tests/test_main.py

```diff
@@ -107,3 +108,17 @@ async def app(scope, receive, send):
     with pytest.raises(SystemExit) as exit_exception:
         run(app, lifespan="on")
     assert exit_exception.value.code == 3
+
+
+def test_run_match_config_params() -> None:
+    config_params = {
+        key: repr(value)
+        for key, value in inspect.signature(Config.__init__).parameters.items()
+        if key not in ("self", "timeout_notify", "callback_notify")
```

Comment:
`timeout_notify` and `callback_notify` are not available via `run()`. Should `run() support it?

Reply:
> `timeout_notify` and `callback_notify` are not available via `run()`. Should `run() support it?

certainly not imho  `callback_notify` it is just there to let the gunicorn know the worker is still alive iirc every `timeout_notify`
those can be overridden if you are subclassing UvicornWorker, I wouldn't know how to do that with the CLI nor if it makes sense


> Probably good to make a constant with those two values `CONFIG_ONLY_PARAMS`

why not yes, not opposed having them that way too

Reply:
> certainly not imho callback_notify it is just there to let the gunicorn know the worker is still alive iirc every timeout_notify
those can be overridden if you are subclassing UvicornWorker, I wouldn't know how to do that with the CLI nor if it makes sense

Right. Thanks 🙏

Reply:
I don't see much benefit on the `CONFIG_ONLY_PARAMS` constant for a single test function. Is there? 

I'll merge this, but I don't mind creating a PR to add that. 🙏